mtcpclient/reporting.go

```diff
@@ -8,10 +8,13 @@ import (
 	"github.com/dachad/tcpgoon/tcpclient"
 )
 
-func collectConnectionsStatus(connectionsStatusRegistry GroupOfConnections, statusChannel <-chan tcpclient.Connection) {
+func collectConnectionsStatus(connectionsStatusRegistry *GroupOfConnections, statusChannel <-chan tcpclient.Connection) {
 	for {
 		newConnectionStatusReported := <-statusChannel
-		connectionsStatusRegistry[newConnectionStatusReported.ID] = newConnectionStatusReported
+		connectionsStatusRegistry.connections[newConnectionStatusReported.ID] = newConnectionStatusReported
+		if len(connectionsStatusRegistry.getFilteredListByStatus([]tcpclient.ConnectionStatus{tcpclient.ConnectionEstablished})) > connectionsStatusRegistry.metrics.maxConcurrentEstalished {
```

Comment:
rather than counting again all the connections in an specific status, maybe you can track the current number in parallel and update it using the new "message" you got...

Reply:
yup, some refactor is coming...
it was the first try, so room for improving ;)